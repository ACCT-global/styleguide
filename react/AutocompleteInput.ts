export { default } from './components/Autocomplete'
