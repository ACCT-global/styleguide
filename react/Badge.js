import Tag from './components/Tag/index'
import deprecated from './modules/deprecated'

export default deprecated(
  {
    useNewComponent: {
      Badge: 'Tag',
    },
  },
  Tag
)
