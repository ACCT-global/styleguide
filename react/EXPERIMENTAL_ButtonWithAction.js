import EXPERIMENTAL_ButtonWithAction from './components/EXPERIMENTAL_ButtonWithAction/index'

export default EXPERIMENTAL_ButtonWithAction
