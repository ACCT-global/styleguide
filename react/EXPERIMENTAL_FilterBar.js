import EXPERIMENTAL_FilterBar from './components/EXPERIMENTAL_FilterBar/index'

export default EXPERIMENTAL_FilterBar
