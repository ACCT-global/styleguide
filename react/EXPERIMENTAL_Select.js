import EXPERIMENTAL_Select from './components/EXPERIMENTAL_Select/index'

export default EXPERIMENTAL_Select
