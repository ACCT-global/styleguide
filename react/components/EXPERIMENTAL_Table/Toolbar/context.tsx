import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from '../types'

const ToolbarContext = createContext<E2ETestable>(null)

export const ToolbarProvider: FC<E2ETestable> = ({ children, testId }) => {
  return (
    <ToolbarContext.Provider value={{ testId }}>
      {children}
    </ToolbarContext.Provider>
  )
}

export function useToolbarContext() {
  const context = useContext(ToolbarContext)
  if (!context) {
    throw new Error('Do not use Toolbar composites outside of context')
  }
  return context
}
