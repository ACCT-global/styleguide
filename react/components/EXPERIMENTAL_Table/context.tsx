import React, { FC, useContext, createContext } from 'react'

import { E2ETestable } from './types'
import useTableMeasures from './hooks/useTableMeasures'

type Measures = ReturnType<typeof useTableMeasures>

const TestingContext = createContext<E2ETestable>(null)
const MeasuresContext = createContext<Measures>(null)

type TableType = E2ETestable & {
  measures: Measures
}

export const TableProvider: FC<TableType> = ({
  children,
  testId,
  measures,
}) => {
  return (
    <TestingContext.Provider value={{ testId }}>
      <MeasuresContext.Provider value={measures}>
        {children}
      </MeasuresContext.Provider>
    </TestingContext.Provider>
  )
}

export function useTestingContext() {
  const context = useContext(TestingContext)
  if (!context) {
    throw new Error(
      'Do not use E2ETestable components outside of the TestingContext'
    )
  }
  return context
}

export function useMeasuresContext() {
  const context = useContext(MeasuresContext)
  if (!context) {
    throw new Error(
      'Do not use measurable components outside of the MeasuresContext'
    )
  }
  return context
}
