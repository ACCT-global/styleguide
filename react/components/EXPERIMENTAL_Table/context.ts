import { createContext } from 'react'

const Context = createContext<TableState>(null)
const { Provider } = Context

export default Context
export { Provider as TableProvider }
