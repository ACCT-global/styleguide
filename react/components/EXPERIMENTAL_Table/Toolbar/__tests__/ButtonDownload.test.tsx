import React from 'react'
import { render } from '@testing-library/react'

import { ButtonGroupProvider } from '../context'
import ButtonDownload from '../ButtonDownload'

describe('Table V2 @ Toolbar/ButtonDownload spec', () => {
  const basicProps = {
    label: 'Download',
    onClick: () => {},
  }

  it('renders the component', () => {
    const button = render(
      <ButtonGroupProvider testId="testId">
        <ButtonDownload {...basicProps} />
      </ButtonGroupProvider>
    )
    expect(button.asFragment()).toMatchSnapshot()
  })
})
