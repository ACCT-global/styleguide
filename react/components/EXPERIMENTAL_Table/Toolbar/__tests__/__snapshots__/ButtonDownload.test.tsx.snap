// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table V2 @ Toolbar/ButtonDownload spec renders the component 1`] = `
<DocumentFragment>
  <div
    class="relative mh2"
    data-testid="testId__download__container"
    id="vtex-table-v2__toolbar__button-download__container"
  >
    <button
      class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
      data-testid="testId__download"
      tabindex="0"
      type="button"
    >
      <div
        class="vtex-button__label flex items-center justify-center h-100 ph5 "
        style="padding-top: .25em; padding-bottom: .32em;"
      >
        <span
          class="flex items-center nr1"
        >
          <div
            class="mr2 nl2 flex items-center"
          >
            <span
              class="c-on-base"
              style="margin-top: 1.5px;"
            >
              <svg
                class="vtex__icon-download  "
                fill="none"
                height="13"
                viewBox="0 0 14 16"
                width="11.375"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <path
                  d="M6.4 12C6.7 12 6.9 11.9 7.1 11.7L12.8 6L11.4 4.6L7.4 8.6V0H5.4V8.6L1.4 4.6L2.38419e-08 6L5.7 11.7C5.9 11.9 6.1 12 6.4 12Z"
                  fill="currentColor"
                  transform="translate(0.599976)"
                />
                <path
                  d="M14 0H0V2H14V0Z"
                  fill="currentColor"
                  transform="translate(0 14)"
                />
              </svg>
            </span>
          </div>
          <div>
            <span
              class="c-on-base"
            >
              Download
            </span>
          </div>
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
